sky/clouds/utils/scp_utils.py

```diff
@@ -442,3 +442,184 @@ def start_instance(self, vm_id):
     def stop_instance(self, vm_id):
         url = f'{API_ENDPOINT}/virtual-server/v2/virtual-servers/{vm_id}/stop'
         return self._post(url=url, request_body={})
+
+    def list_security_group_rules(self, sg_id):
+        url = f'{API_ENDPOINT}/security-group/v2/security-groups/{sg_id}/rules'
+        return self._get(url)
+
+    def _check_existing_security_group_rule(self, sg_id, direction, ports):
```

Comment:
```suggestion
    def _check_existing_security_group_rule(self, sg_id, direction, ports):
```
this is actually "check whether it is non-exist"? should we revert the bool value or change the function name?

Reply:
this function is to check whether it is exist. if you want to change something, please let me know

Reply:
nit: I mean the function name should indicate the return value. e.g. `check_rule_not_exist`'s return value (a boolean value), if it is true, then it should mean the rule *does not exist*.